### ISSUE TITLE
Unnamed colormaps adding is not thread safe

### ISSUE THREAD

**Issue by Czaki** [napari/napari#2182]
## 🐛 Bug
I got an error report from my application. 

```
RuntimeError: dictionary changed size during iteration
  File "napari\_qt\qthreading.py", line 528, in reraise
  File "napari\_qt\qthreading.py", line 124, in run
  File "napari\_qt\qthreading.py", line 224, in work
  File "PartSeg\common_gui\napari_image_view.py", line 713, in _prepare_layers
  File "napari\layers\image\image.py", line 264, in __init__
  File "napari\layers\intensity_mixin.py", line 54, in colormap
  File "napari\utils\colormaps\colormap_utils.py", line 426, in ensure_colormap
  File "napari\utils\colormaps\colormap_utils.py", line 376, in increment_unnamed_colormap
  File "napari\utils\colormaps\colormap_utils.py", line 376, in <listcomp>
```

So it happens in line https://github.com/napari/napari/blob/58677b5a94a19382cbfa0e743a8ee5bbc59aaac7/napari/utils/colormaps/colormap_utils.py#L376

and place in my code which causes it is here https://github.com/4DNucleome/PartSeg/blob/928ec8c8d40479ddca4ab8cb1381730d41a4bcc6/package/PartSeg/common_gui/napari_image_view.py#L713

I see that I could ensure registration on my side, but maybe this section in code should be protected by some mutex/semaphore?

I could provide more information about variables, but I only have a Sentry report.

## Expected behavior

This part of the code should be safe for threading. 

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

 - Any other relevant information:

## Additional context

If there is a decision to fix and how to fix I could prepare PR.

**Comment by sofroniewn** [comment]
Hmm agreed this feels like it should be fixed. I don't have ideas/ strong opinions on what is best, curious if others do. If it's easy maybe you can make a PR / pseudo-code with a couple of the different potential fix options and then we decide? Unless others have good ideas on how to approach

**Comment by Czaki** [comment]
I have created #2193 PR. This is the simples change. Another possibility is to refactor the function to put fragile parts in one place.

**Comment by sofroniewn** [comment]
Closed by #2193
